From 0ecaaf9098dc22b5503dfbce5cf629525e1dd237 Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Fri, 7 May 2021 14:30:03 +0800
Subject: [PATCH] plugin_api: fix build with gcc 11

Some C++ Standard Library headers have been changed in gcc 11 to no
longer include other headers that they do need to depend on.

Include exception explicitly to avoid:

| inference-engine/src/plugin_api/ie_system_conf.h:21:31: error: 'exception_ptr' in namespace 'std' does not name a type; did you mean 'exception'?
|    21 | INFERENCE_ENGINE_API_CPP(std::exception_ptr&) CurrentException();
|      |                               ^~~~~~~~~~~~~

Upstream-Status: Submitted [https://github.com/openvinotoolkit/openvino/pull/5546]

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
---
 inference-engine/src/plugin_api/ie_system_conf.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/inference-engine/src/plugin_api/ie_system_conf.h b/inference-engine/src/plugin_api/ie_system_conf.h
index c0e4cbb8b..56d5fb488 100644
--- a/inference-engine/src/plugin_api/ie_system_conf.h
+++ b/inference-engine/src/plugin_api/ie_system_conf.h
@@ -11,6 +11,7 @@
 
 #include "ie_api.h"
 #include <vector>
+#include <exception>
 
 namespace InferenceEngine {
 
-- 
2.30.2

